package/MDAnalysis/analysis/pca.py

```diff
@@ -203,11 +208,13 @@ def transform(self, atomgroup, n_components=None):
         if self._atoms != atomgroup:
             warnings.warn('This is a transform for different atom types.')
 
-        xyz = np.array([atomgroup.positions.copy() for ts in self._u.trajectory])
-        xyz = xyz.reshape(self._u.trajectory.n_frames,
-                          self._n_atoms*3, order='F')
+        dot = np.zeros((u.trajectory.n_frames, ca.n_atoms*3))
+
+        for i, ts in enumerate(self._u.trajectory[self.start:self.stop:self.step]):
+            xyz = atomgroup.positions.ravel()
+            dot[i] = np.dot(xyz, self.p_components[:n_components].T)
 
-        return np.dot(xyz, self.p_components[:n_components].T)
+        return dot
 
     def inverse_tranform(self, pca_space):
```

Comment:
maybe we can over to write a trajectory with the new coordinates as well? This could be done by adding a flag here or adding a new function. 

Having a function to that write a trajectory from an atomgroup and array will actually be generally useful. Can you please open a new issue for that. We can talk there where it should be put and you can implement it in another small PR.